packages/autocomplete-plugin-recent-searches/src/createRecentSearchesPlugin.ts

```diff
@@ -64,13 +64,22 @@ export function createRecentSearchesPlugin<TItem>({
       ];
     },
     onSubmit: ({ state }) => {
-      store.add({
-        objectID: state.query,
-        query: state.query,
-      });
+      const { query } = state;
+      if (query) {
+        store.add({
+          objectID: query,
+          query,
+        });
+      }
     },
     onSelect: ({ suggestion }) => {
-      store.add(suggestion);
+      const { query, objectID } = suggestion as any;
+      if (query && objectID) {
```

Comment:
This won't work with non-Algolia records so we should rather store the record and fallback `objectID` to `query`.

Reply:
@francoischalifour Okay, that part makes sense.

However I'm gonna have to leave `if (query)` condition.
For example,
![image](https://user-images.githubusercontent.com/499898/93600019-564a4880-f9bf-11ea-88f0-9d1f858ae14c.png)
these products (not coming from suggestion index) don't have `query` attribute.
In this case, is it correct that we're not storing them? Not sure how we've decided, but if it's correct, we need to keep `if (query)` clause. cc @Shipow

Reply:
Perhaps this deserves a sync on Monday?

Reply:
https://github.com/algolia/autocomplete.js/blob/feat/recent-searches/packages/autocomplete-core/src/getPropGetters.ts#L323-L336

If the source has `getSuggestionUrl`, it doesn't reach to `ource.onSelect`, so it naturally doesn't get to be saved in recent searches.